enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -164,71 +204,45 @@ pinned_images=[
 ]
 ```
 
-In addition to the list of images to be pinned, the `PinnedImageSet` custom
-resource will also contain a node selector. This is intended to support
-different sets of images for different kinds of nodes. For example, to pin
-different images for control plane and worker nodes the user could create two
-`PinnedImageSet` custom resources:
-
-```yaml
-# For control plane nodes:
-apiVersion: machineconfiguration.openshift.io/v1alpha1
-kind: PinnedImageSet
-metadata:
-  name: my-control-plane-pinned-images
-spec:
-  nodeSelector:
-    matchLabels:
-      node-role.kubernetes.io/control-plane: ""
-  pinnedImages:
-  ...
-
----
-
-# For worker nodes:
-apiVersion: machineconfiguration.openshift.io/v1alpha1
-kind: PinnedImageSet
-metadata:
-  name: my-worker-pinned-images
-spec:
-  nodeSelector:
-    matchLabels:
-      node-role.kubernetes.io/worker: ""
-  pinnedImages:
-  ...
-```
+The `PinnedImageSet` is tightly coupled to `MachineConfigPool` and each `CR` can
+be referenced on the pool level via the `MachineConfigPoolSpec`. This is
+flexible becuase it allows a controller/admin the ability to prescribe sets to
+pools. Although a cluster only has two pools by default `master` and `worker`
+[custom pools](https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md)
+can be added for further fine grained controls.
 
 This is specially convenient for pinning images for upgrades: there are many
 images that are needed only by the control plane nodes and there is no need to
 have them consuming disk space in worker nodes.
 
-When no node selector is specified the images will be pinned in all the nodes
-of the cluster.
-
-The _machine-config-controller_ will grow a new `PinnedImageSetController` sub
-controller that will watch the `PinnedImageSet` custom resources. When one of
-those is created, updated or deleted the controller will start a daemon set that
-will do the following in each node of the cluster:
+The _machine-config-daemon_ will grow a new `prefetch_image_manager` utilizing
+the same general flow as the `certificate_writer` which is not dependent on
+defining the configuration as `MachineConfig`.  controller that will watch the
+`PinnedImageSet` and `MachineConfigPool` resources. On sync the manager will
+perform the following tasks.
+
+1. Using the same mechanism as `MachineConfigDaemon` update() flow
```

Comment:
this sounds great. Let's just ensure that its clear where the error is coming from in the MCD. You would propagate an err msg up using the nodeWriter, right?

Reply:
correct